gltf: Add GLTFHandleBinary::HANDLE_BINARY_EMBED_AS_UNCOMPRESSED

lyuma opened the pull request:
This option allows for a safe fallback for embedded gltf textures in cases where VRAM compression is not needed.
Add an is_editor_hint guard around GLTFHandleBinary::HANDLE_BINARY_EXTRACT_TEXTURES, to use EMBED_AS_UNCOMPRESSED by default at runtime.
This provides an option for pixel art to be stored losslessly.
Additionally, respect project importer defaults for texture import settings.
Threw in another check to avoid overwriting and reimporting textures which were already extracted & identical to what is on disk.

![editor import settings showing new embed as uncompressed option](https://user-images.githubusercontent.com/39946030/215698810-725adb97-0e08-4eba-b506-3bb457e20feb.png)

Fixes #72200
Fixes #72195
Fixes #72467
Fixes #72191

lyuma committed:
gltf: Add GLTFHandleBinary::HANDLE_BINARY_EMBED_AS_UNCOMPRESSED

This option allows for a safe fallback for embedded gltf textures in cases where VRAM compression is not needed.
Add an is_editor_hint guard around GLTFHandleBinary::HANDLE_BINARY_EXTRACT_TEXTURES, to use EMBED_AS_UNCOMPRESSED by default at runtime.
This provides an option for pixel art to be stored losslessly.
Additionally, respect project importer defaults for texture import settings.
Avoid writing and reimporting extracted textures identical to version on disk.

akien-mga commented on modules/gltf/gltf_state.h:
```diff
@@ -108,6 +108,7 @@ class GLTFState : public Resource {
 		HANDLE_BINARY_DISCARD_TEXTURES = 0,
 		HANDLE_BINARY_EXTRACT_TEXTURES,
 		HANDLE_BINARY_EMBED_AS_BASISU,
+		HANDLE_BINARY_EMBED_AS_UNCOMPRESSED,
```
Probably needs a comment that if the enum value changes, the 3 hardcore in core needs to be changed too.

lyuma replied:
I'll add a comment. It's still a bit icky

reduz commented on modules/gltf/editor/editor_scene_importer_gltf.cpp:
```diff
@@ -87,7 +87,7 @@ Node *EditorSceneFormatImporterGLTF::import_scene(const String &p_path, uint32_t
 void EditorSceneFormatImporterGLTF::get_import_options(const String &p_path,
 		List<ResourceImporter::ImportOption> *r_options) {
-	r_options->push_back(ResourceImporterScene::ImportOption(PropertyInfo(Variant::INT, "meshes/handle_gltf_embedded_images", PROPERTY_HINT_ENUM, "Discard All Textures,Extract Textures,Embed As Basis Universal", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_UPDATE_ALL_IF_MODIFIED), GLTFState::HANDLE_BINARY_EXTRACT_TEXTURES));
+	r_options->push_back(ResourceImporterScene::ImportOption(PropertyInfo(Variant::INT, "meshes/handle_gltf_embedded_images", PROPERTY_HINT_ENUM, "Discard All Textures,Extract Textures,Embed As Basis Universal,Embed as Uncompressed", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_UPDATE_ALL_IF_MODIFIED), GLTFState::HANDLE_BINARY_EXTRACT_TEXTURES));
```
Please add a GLTF section for this, such as:
`gltf/embedded_image_handling` instead of taking over the mesh section.

lyuma replied:
I changed it to use this name

lyuma commented:
Updated the PR to also respect project import defaults. It will reimport them with Detect 3D after the gltf completes, which should be okay since it will run multithreaded, and avoids some hardcoded constants.

The original code set "compress/mode" to 2 but wrote remap instead of params, so it didn't actually work. Now that I'm reading the project defaults, there isn't a clean way to override the compression mode without disregarding the user's project settings, so I removed that part.

akien-mga commented:
Thanks!
